posthog/temporal/batch_exports/http_batch_export.py

```diff
@@ -0,0 +1,251 @@
+                    }
+
+                    write_event_to_batch(capture_event)
+
+                    if (
+                        batch_file.tell() > settings.BATCH_EXPORT_HTTP_UPLOAD_CHUNK_SIZE_BYTES
+                        or batch_file.records_since_last_reset >= settings.BATCH_EXPORT_HTTP_BATCH_SIZE
+                    ):
+                        await flush_batch()
+                        batch_file.reset()
+
+            if batch_file.tell() > 0:
+                await flush_batch()
+
+
+@workflow.defn(name="http-export")
+class HttpBatchExportWorkflow(PostHogWorkflow):
+    """A Temporal Workflow to export ClickHouse data to an HTTP endpoint.
+
+    This Workflow is intended to be executed both manually and by a Temporal
+    Schedule. When ran by a schedule, `data_interval_end` should be set to
+    `None` so that we will fetch the end of the interval from the Temporal
+    search attribute `TemporalScheduledStartTime`.
+    """
+
+    @staticmethod
+    def parse_inputs(inputs: list[str]) -> HttpBatchExportInputs:
+        """Parse inputs from the management command CLI."""
+        loaded = json.loads(inputs[0])
+        return HttpBatchExportInputs(**loaded)
+
+    @workflow.run
+    async def run(self, inputs: HttpBatchExportInputs):
+        """Workflow implementation to export data to an HTTP Endpoint."""
+        data_interval_start, data_interval_end = get_data_interval(inputs.interval, inputs.data_interval_end)
+
+        create_export_run_inputs = CreateBatchExportRunInputs(
+            team_id=inputs.team_id,
+            batch_export_id=inputs.batch_export_id,
+            data_interval_start=data_interval_start.isoformat(),
+            data_interval_end=data_interval_end.isoformat(),
+        )
+        run_id = await workflow.execute_activity(
+            create_export_run,
+            create_export_run_inputs,
+            start_to_close_timeout=dt.timedelta(minutes=5),
+            retry_policy=RetryPolicy(
+                initial_interval=dt.timedelta(seconds=10),
+                maximum_interval=dt.timedelta(seconds=60),
+                maximum_attempts=0,
+                non_retryable_error_types=[],
+            ),
+        )
+
+        update_inputs = UpdateBatchExportRunStatusInputs(
+            id=run_id,
+            status="Completed",
+            team_id=inputs.team_id,
+        )
+
+        insert_inputs = HttpInsertInputs(
+            team_id=inputs.team_id,
+            url=inputs.url,
+            token=inputs.token,
+            data_interval_start=data_interval_start.isoformat(),
+            data_interval_end=data_interval_end.isoformat(),
+            exclude_events=inputs.exclude_events,
+            include_events=inputs.include_events,
+            batch_export_schema=inputs.batch_export_schema,
+        )
+
+        await execute_batch_export_insert_activity(
+            insert_into_http_activity,
+            insert_inputs,
+            non_retryable_error_types=[],
```

Comment:
Might be worth it to not leave this empty, for example, 400 errors should be non-retryable? (except for 429).